app/code/Magento/Inventory/Model/StockRepository.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
```

Comment:
we need to have delete()
operation in Stock repository.

When Stock is removed, we need to remove all stock items related to it (could be done using Foreign Key on DB level)